contracts/tokens/IpToken.sol

```diff
@@ -41,7 +41,11 @@ contract IpToken is IporOwnable, IIpToken, ERC20 {
         return _asset;
     }
 
-    function setJoseph(address newRouter) external override onlyOwner {
+    function getRouter() external view override returns (address) {
```

Comment:
we can not change this name
it is on blockchain

Reply:
I suggest to use new approach in new version. Based on our consultation I changed to setTokenManager